packages/persistent-merkle-tree/src/node.ts

```diff
@@ -126,3 +354,39 @@ export function compose(inner: Link, outer: Link): Link {
     return outer(inner(n));
   };
 }
+
+export function getNodeH(node: Node, hIndex: number): number {
+  if (hIndex === 0) return node.h0;
```

Comment:
will the switch here be faster? also not sure how hot this path is.

Reply:
Extremely hot, I did benchmarks with different options:
- if
- switch
- array to property

And if + switch had similar speeds. Makes sense because it is fundamentally the same operation. Both if and switch check every possible case before returning. Chose if because it looks more readable and concise.